nano/node/node.cpp

```diff
@@ -3477,9 +3519,94 @@ void nano::active_transactions::update_difficulty (nano::block const & block_a)
 		uint64_t difficulty;
 		auto error (nano::work_validate (block_a, &difficulty));
 		assert (!error);
-		roots.modify (existing, [difficulty](nano::conflict_info & info_a) {
-			info_a.difficulty = difficulty;
-		});
+		if (difficulty > existing->difficulty)
+		{
+			roots.modify (existing, [difficulty](nano::conflict_info & info_a) {
+				info_a.difficulty = difficulty;
+			});
+			adjust_difficulty (block_a.hash ());
+		}
+	}
+}
+
+void nano::active_transactions::adjust_difficulty (nano::block_hash const & hash_a)
+{
+	assert (!mutex.try_lock ());
+	std::deque<std::pair<nano::block_hash, int64_t>> remaining_blocks;
+	remaining_blocks.push_back (std::make_pair (hash_a, 0));
+	std::unordered_set<nano::block_hash> processed_blocks;
+	std::vector<std::pair<nano::uint512_union, int64_t>> elections_list;
+	uint128_t sum (0);
+	while (!remaining_blocks.empty ())
+	{
+		auto item (remaining_blocks.front ());
+		remaining_blocks.pop_front ();
+		auto hash (item.first);
+		// item.second is level
```

Comment:
perhaps use a variable named `level` instead of this comment

Reply:
Thx